packages/modules/itwin-viewer-react/src/components/app-ui/widgets/BasicNavigationWidget.tsx

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** Clone of core BasicNavigationWidget with conditional tooling
+ */
+import {
+  CommonToolbarItem,
+  ToolbarOrientation,
+  ToolbarUsage,
+} from "@bentley/ui-abstract";
+import {
+  CoreTools,
+  NavigationWidgetComposer,
+  ToolbarComposer,
+  ToolbarHelper,
+} from "@bentley/ui-framework";
+import * as React from "react";
+
+import { ViewNavigationTools } from "../../../types";
+import { useUiVisibility } from "./BasicToolWidget";
+
+export interface BasicNavigationWidgetProps {
+  config?: ViewNavigationTools;
+}
+
+/** Basic Navigation Widget that provides standard tools to manipulate views containing element data.
+ * Supports the specification of additional horizontal and vertical toolbar items through props.
+ */
+export function BasicNavigationWidget({ config }: BasicNavigationWidgetProps) {
+  const getHorizontalToolbarItems = React.useCallback((): CommonToolbarItem[] => {
+    if (config?.hideDefaultHorizontalItems) {
+      return [];
+    }
+    const items: CommonToolbarItem[] = ToolbarHelper.createToolbarItemsFromItemDefs(
```

Comment:
Everything looks good to me, except just for curiosity, I wonder if is it okay to use widgets e.g BasicNavigationWIdget from ui-framework? I am not sure if ui-framework is open source or not, however, I remember Bill Steinbock's presentation, where he said ui-framework provides most of the widget we need and reduces all these repetition. Also, I haven't verified whether widgets provided by Ui-Framework are hideable or not, but they sure are responsive.

Reply:
That's what I am replacing with this one. This is basically the same widget but it adds a prop for configuration to hide the default tools. The core version does not allow this. It only allows expansion